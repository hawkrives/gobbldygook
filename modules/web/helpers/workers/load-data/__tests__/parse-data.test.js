/* eslint-env jest */
// @flow

import parseData from '../parse-data'

test('parseData can parse json', () => {
    expect(parseData('{"foo": 2}', 'courses')).toMatchSnapshot()
})

test('parseData can parse yaml', () => {
    expect(parseData('foo: 2', 'areas')).toMatchSnapshot()
})

test("parseData returns a blank object if it can't parse", () => {
    // $FlowFixMe
    expect(parseData('foo: 2', 'other')).toMatchSnapshot()
    expect(parseData('invalid', 'courses')).toMatchSnapshot()
    expect(parseData('- invalid: yaml:', 'areas')).toMatchSnapshot()
})
