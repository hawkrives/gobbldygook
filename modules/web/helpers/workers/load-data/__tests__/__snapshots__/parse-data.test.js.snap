// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseData can parse json 1`] = `
Object {
  "foo": 2,
}
`;

exports[`parseData can parse yaml 1`] = `
Object {
  "foo": 2,
  "source": "foo: 2",
}
`;

exports[`parseData returns a blank object if it can't parse 1`] = `Object {}`;

exports[`parseData returns a blank object if it can't parse 2`] = `Object {}`;

exports[`parseData returns a blank object if it can't parse 3`] = `Object {}`;
